app/src/common/shared/com/igalia/wolvic/VRBrowserActivity.java

```diff
@@ -1944,6 +1944,34 @@ public void openNewTabForeground(@NonNull String uri) {
         mWindows.addTab(mWindows.getFocusedWindow(), uri);
     }
 
+    @Override
+    public void openNewPageNoInterrupt(@NonNull String uri) {
+        Session focusedSession = mWindows.getFocusedWindow().getSession();
+        if (focusedSession != null && focusedSession.getActiveVideo() == null) {
+            // No video playing in the current window, we are good to go!
+            openNewTabForeground(uri);
+            return;
+        }
+        // If we have video playing in current window, ensure we don't open a new tab
+        // in a window that has active video
+        if (mWindows.getWindowsCount() > 1) {
+            for (WindowWidget window : mWindows.getCurrentWindows()) {
+                if (window.getSession() != null && window.getSession().getActiveVideo() == null) {
```

Comment:
Consider refactoring this check in a private method so it can be used also in the other one above

Reply:
Done!